CRM/Member/Form/MembershipRenewal.php

```diff
@@ -497,20 +561,24 @@ protected function submit() {
     $this->assign('receive_date', $this->_params['receive_date']);
     $this->processBillingAddress();
     list($userName) = CRM_Contact_BAO_Contact_Location::getEmailDetails(CRM_Core_Session::singleton()->get('userID'));
-    $this->_params['total_amount'] = CRM_Utils_Array::value('total_amount', $this->_params,
-      CRM_Core_DAO::getFieldValue('CRM_Member_DAO_MembershipType', $this->_memType, 'minimum_fee')
-    );
-    $this->_membershipId = $this->_id;
-    $customFieldsFormatted = CRM_Core_BAO_CustomField::postProcess($this->_params,
-      $this->_id,
-      'Membership'
-    );
-    if (empty($this->_params['financial_type_id'])) {
-      $this->_params['financial_type_id'] = CRM_Core_DAO::getFieldValue('CRM_Member_DAO_MembershipType', $this->_memType, 'financial_type_id');
+    $this->assign('contactID', $this->_contactID);
+
+    if ($this->_priceSet) {
+      // renewing using a price set, set the default financial type (if not set) to the price set's default financial type.
+      if (empty($this->_params['financial_type_id'])) {
```

Comment:
We don't need the empty check - one of the few differences between $quickConfig and normal price sets is that for quick config it is possible to override the financial_type_id